block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -308,7 +309,7 @@ impl BlockWitness {
             merkle_store
                 .add_merkle_paths(self.produced_nullifiers.iter().map(|(nullifier, proof)| {
                     // Note: the initial value for all nullifiers in the tree is `[0, 0, 0, 0]`
-                    (u64::from(nullifier[3]), Digest::default(), proof.path().clone())
+                    (u64::from(nullifier.inner()[3]), Digest::default(), proof.path().clone())
```

Comment:
Ideally, instead of doing `nullifier.inner()[3]` we should add a method on the `Nullifier` to get the most significant element. Maybe something like `Nullifier::mso()` or `Nullifier::most_significant_felt()`?